src/looprestoration_tmpl_16.rs

```diff
@@ -353,15 +186,15 @@ unsafe extern "C" fn sgr_5x5_rust(
     let mut tmp: [pixel; 27300] = [0; 27300];
     let mut dst: [coef; 24576] = [0; 24576];
     padding::<BitDepth16>(&mut tmp, p, stride, left, lpf, w, h, edges);
-    selfguided_filter(
+    selfguided_filter::<BitDepth16>(
```

Comment:
```suggestion
    selfguided_filter(
```

Shouldn't be needed since you specify `BitDepth16::from_c(bitdepth_max)` as an arg.

Reply:
Done!